draft-ietf-mls-protocol.md

```diff
@@ -1161,43 +1175,15 @@ in a similar fashion. This allows forward secrecy at the level of
 application messages within and out of an epoch.
 A step in this chain (the second subscript) is called a "generation".
 
-~~~~~
-           application_secret
-                     |
-                     V
-           HKDF-Expand-Label(., "app sender", [sender], Hash.length)
-                     |
-                     V
-           application_secret_[sender]_[0]
-                     |
-                    ...
-                     |
-                     V
-           application_secret_[sender]_[N-1]
-                     |
-                     +--> HKDF-Expand-Label(.,"nonce", "", nonce_length)
-                     |    = write_nonce_[sender]_[N-1]
-                     |
-                     +--> HKDF-Expand-Label(.,"key", "", key_length)
-                     |    = write_key_[sender]_[N-1]
-                     V
-           HKDF-Expand-Label(., "app sender", [sender], Hash.length)
-                     |
-                     V
-           application_secret_[sender]_[N]
-~~~~~
-
-As before the value [sender] represents the index of the member that will
-use this key to send, encoded as a uint32.
-
 [[ OPEN ISSUE: The HKDF context field is left empty for now.
 A proper security study is needed to make sure that we do not need
 more information in the context to achieve the security goals.]]
```

Comment:
It seems like this OPEN ISSUE is obsoleted by the new key schedule.

Reply:
Actually, I would just replace the whole discussion of application keys here with a forward reference to the application key schedule section.  All you need here is the stuff that's required to underestand the message framing, namely the idea of "generations".  So maybe just put a short summary, e.g.:

> For application messages, a chain of keys is derived for each sender in a similar fashion. This allows forward secrecy at the level of application messages within and out of an epoch.  A step in this chain (the second subscript) is called a "generation". The details of application key derivation are described in {{message-protection}}.

And delete the remainder of this section.